homeassistant/components/homekit_controller/__init__.py

```diff
@@ -214,4 +230,9 @@ def get_characteristic_types(self):
 async def async_remove_entry(hass, entry):
```

Comment:
Why are we using `async_remove_entry` and not `async_unload_entry`?

Side note, `async_unload_entry` is called as part of removing an entry.

Reply:
Good spot. Had just conflated the 2, really deleting the entity map is a job for remove_entry but you are right the other stuff belongs in unload.

Have split them out and added a test to make sure its called. Switched to using `MockConfigEntry.async_setup` so I could use `MockConfigEntry.async_unload` - this caused a bit of faff updating test fixtures but I think its worth it.